.github/workflows/test.yml

```diff
@@ -0,0 +1,82 @@
+on:
+  - push
+
+jobs:
+  test-clj:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3.0.2
+
+      - name: Setup Java
+        uses: actions/setup-java@v3.4.0
+        with:
+          distribution: temurin
+          java-version: 17
+
+      - name: Setup Clojure
+        uses: DeLaGuardo/setup-clojure@9.3
+        with:
+          cli: latest
+          lein: latest
+          bb: latest
+
+      - name: Cache
+        uses: actions/cache@v3.0.5
+        with:
+          path: |
+            ~/.m2/repository
+            ~/.gitlibs
+            ~/.deps.clj
+          key: cljdeps-${{ hashFiles('project.clj', 'deps.edn', 'bb.edn') }}
+          restore-keys: cljdeps-
+
+      - name: Uberjar
+        run: lein uberjar
+
+      - name: Babashka tests
+        run: bb test:bb
+
+      - name: Clojure tests
+        run: clojure -M:cljtest:humane:runner
+
+      - name: Integration tests
+        run: ./test_config 1.2.3 uberjar
+
+  test-cljs:
+    runs-on: macos-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3.0.2
+
+      - name: Setup Java
+        uses: actions/setup-java@v3.4.0
+        with:
+          distribution: temurin
+          java-version: 17
+
+      - name: Setup Clojure
+        uses: DeLaGuardo/setup-clojure@9.3
+        with:
+          cli: latest
+          lein: latest
+          bb: latest
```

Comment:
Have you considered using fixed versions? It's often quite annoying to have CI builds break for seemingly no reason

Reply:
Yes, and actually I would like to propose using dependency updating the tool I'm actively contributing to: https://github.com/renovatebot/renovate

The app is free for GitHub, and it handles `project.clj` as well as `deps.edn`.

Using it, we would have both of the two worlds: fresh dependencies (thanks to Renovate) and confidence during merge (thanks to CI tests)

Reply:
Here is my demo of what is possible with automatic dependency updates:
https://github.com/zharinov/zprint/pulls

@kkinnear What do you think?

Reply:
Wow, that's amazing.  As you may have noticed, I rarely change the versions of the things on which zprint depends, as I find it a pain to debug things like that when they break.  Not that they break all that often, other than graalVM, which I noticed isn't something that `renovate` found.  The thought of having major dependencies automatically updated (or at least automatically PR'ed for update) is kind of scary, but probably a good idea.  As long as we really believe that the automated tests will run.

I didn't read too closely, but is it the case that if it automatically accepts the `renovate` PR's, and the tests don't run correctly, then it will back them out or something?   Or is that a manual operation?

In any case, I'm up for giving `renovate` a try, but I think that first we need to get the tests to work in GitHub actions, including `test_config`, and at least the Linux pre-built binary (`zprintl-1.2.n`).  It would be nice to do the Mac version as well, but I expect that is not possible with GitHub actions.  Anyway, once the tests work, then adding `renovate` to the mix would seem like a good, if kind of unsettling, idea.  Thanks for proposing it!

Reply:
It can create PRs for you to review and merge, but also can be configured to automatically merge non-major releases once all CI checks are passing, leaving major updates up to you.

Actually, it's very configurable. For example, I think it's possible to make it upgrade clojure dependencies that are documented in the markdown files (using regex).

So again, recommend you to give it a try once we're done here with build/test pipeline.

Reply:
Speaking of GraalVM, it's also possible to upgrade automatically (and test on CI), as soon as we refer to https://github.com/graalvm/graalvm-ce-builds releases somewhere in the code